module/executiondatasync/pruner/pruner.go

```diff
@@ -184,6 +190,25 @@ func (p *Pruner) loop(ctx irrecoverable.SignalerContext, ready component.ReadyFu
 	}
 }
 
+func (p *Pruner) lowestProducersHeight() uint64 {
+	var heights []uint64
+	for _, producer := range p.registeredProducers {
+		height := producer.LastProcessedHeight()
+		heights = append(heights, height)
+	}
+	sort.Slice(heights, func(i, j int) bool { return heights[i] < heights[j] })
+
+	return heights[0]
```

Comment:
You can get by without the array and sorting by iterating the producers, and tracking the smallest height found.

Reply:
[1b9a7b8](https://github.com/onflow/flow-go/pull/6217/commits/1b9a7b8ed475757da4acd4b9b1cd45408bf022fb)